universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java

```diff
@@ -369,6 +372,36 @@ public ProgramDefinition deleteBlock(long programId, long blockDefinitionId)
     return programMaybe.get().getApplications();
   }
 
+  @Override
+  public void publishProgram(long id) throws ProgramNotFoundException {
+    try {
+      this.publishProgramAsync(id).toCompletableFuture().join();
+    } catch (CompletionException e) {
+      if (e.getCause() instanceof ProgramNotFoundException) {
+        throw new ProgramNotFoundException(id);
+      }
+      throw e;
+    }
+  }
+
+  @Override
+  public CompletionStage<Void> publishProgramAsync(long id) {
+    return programRepository
+        .lookupProgram(id)
+        .thenApplyAsync(
+            programMaybe ->
+                programMaybe.orElseThrow(
+                    () -> new RuntimeException(new ProgramNotFoundException(id))))
+        .thenComposeAsync(programRepository::publishProgramAsync);
```

Comment:
Suggestion (blocking): need to specify the http execution context for the callbacks. Alternatively could delete the async method since an applicant will never execute this code path.